firehose.c

```diff
@@ -437,10 +437,12 @@ static int firehose_erase(struct qdl_device *qdl, struct program *program)
 	return ret == FIREHOSE_ACK ? 0 : -1;
 }
 
-static int firehose_program(struct qdl_device *qdl, struct program *program, int fd)
+static int firehose_program(struct qdl_device *qdl, struct program *program,
+			    int fd, enum qdl_storage_type storage)
```

Comment:
Considering that `struct qdl_device` already has some storage-related fields like `size_t sector_size`, why not include `qdl_storage_type` into this structure as well?

This will remove the need pass one more param to functions like this